src/packit_deploy/config.py

```diff
@@ -84,9 +95,177 @@ def dark_mode_enabled(self) -> bool:
             return self.theme_dark is not None
 
 
+@dataclass
+class PackitAuthGithub:
+    org: str
+    team: str
+    client_id: str
+    client_secret: str
+    oauth2_redirect_packit_api_root: str
+    oauth2_redirect_url: str
+
+    @classmethod
+    def from_data(cls, dat, key: list[str]) -> "PackitAuthGithub":
+        org = config.config_string(dat, [*key, "github_api_org"])
+        team = config.config_string(dat, [*key, "github_api_team"])
+        client_id = config.config_string(dat, [*key, "github_client", "id"])
+        client_secret = config.config_string(dat, [*key, "github_client", "secret"])
+        oauth2_redirect_packit_api_root = config.config_string(dat, [*key, "oauth2", "redirect", "packit_api_root"])
+        oauth2_redirect_url = config.config_string(dat, [*key, "oauth2", "redirect", "url"])
+        return PackitAuthGithub(
+            org=org,
+            team=team,
+            client_id=client_id,
+            client_secret=client_secret,
+            oauth2_redirect_packit_api_root=oauth2_redirect_packit_api_root,
+            oauth2_redirect_url=oauth2_redirect_url,
+        )
+
+
+@dataclass
+class PackitAuth:
+    VALID_AUTH_METHODS = frozenset(("github", "basic", "preauth"))
+
+    method: Union[str, PackitAuthGithub]
```

Comment:
It feels like it might be a little bit nicer if `method` was always a string, and add `method_custom_values` which is `PackitAuthGithub` if method is "github" (and `None` if it's anything else, but with the option to add custom values for the other auth types.